src/configs/recommended.ts

```diff
@@ -63,6 +63,7 @@ export const rules = {
     "max-line-length": {
         options: [120],
     },
+    "max-nesting-depth": true,
```

Comment:
Adding this to `recommended.ts` would be a breaking change. I'll remove it. We can look into adding it back in for the next major release, though I don't know that the importance of enabling this rule trumps the annoyance some users would have seeing errors from it on existing code.